mavros/src/plugins/hil.cpp

```diff
@@ -307,41 +307,39 @@ class HilPlugin : public plugin::PluginBase {
 	 * Message specification: @p https://pixhawk.ethz.ch/mavlink/#HIL_RC_INPUTS_RAW
 	 */
 	void rcin_raw_cb(const mavros_msgs::RCIn::ConstPtr &req) {
-		mavlink::common::msg::HIL_RC_INPUTS_RAW rcin{};
+		mavlink::common::msg::HIL_RC_INPUTS_RAW rcin {};
 
 		constexpr size_t MAX_CHANCNT = 12;
-		size_t channels_count = req->channels.size();
 
-		if (channels_count > MAX_CHANCNT) {
+		if (req->channels.size() > MAX_CHANCNT) {
 			ROS_WARN_THROTTLE_NAMED(60, "hil_rc",
 						"FCU receives %lu RC channels, but HIL_RC_INPUTS_RAW can store %zu",
 						req->channels.size(), MAX_CHANCNT);
-
-			channels_count = MAX_CHANCNT;
 		}
 
+		std::array<uint16_t, MAX_CHANCNT> channels;
+		size_t n = req->channels.size() < MAX_CHANCNT ? : MAX_CHANCNT;
+		std::copy(req->channels.begin(), req->channels.begin() + n, channels.begin());
+		std::fill(channels.begin() + n, channels.end(), UINT16_MAX);
```

Comment:
Or better to use `std::min()` from `#include <algorithm>`.
That template fn require same type for a & b, but anyway size() is always std::size_t.

```cpp
auto n = std::min(req->channels.size(), channels.size());
```